### PR TITLE
Reduce duplicate data from entity columns

### PR DESCRIPTION
**Pull request by Fox32**
I hate that this PR had three features in it, but these matched so well together 😉 If desired we can split it up and land them one by one.

**Change No.1**

I removed the system and domain column in some of the tables as suggested in #4553. @adamdmharvey can you take a look if I missed something? I choose not to modify the presets as they are also used in other places (except the one for the domains, but I would prefer to keep that one as-is, too).

image

image

image

**Change No.2**

I change the description column to take up as much space as possible while keeping the overflow support. That work quite neat when resizing. Both for the catalog and api-docs.

Before (actual this screenshot is wrong, because it's without the new overflow feature...):

image

After:

image

**Change No.3**

Introduced the new overflow component in the description column of the entity table:

image

Here I noticed an issue with the overflow component. Look closely at the "g" in the description. The lowest 1-2 pixel rows are cut off...

image

This is actual an issue with react-text-truncate as it sets the inner span to `overflow: hidden` but I don't get why. If I remove it, it still works fine... We could do it ugly this way, but I'm not sure which other stuff I might break that way:

```diff
+const useStyles = makeStyles({
+  container: {
+    overflow: 'visible !important',
+  },
+});

 export const OverflowTooltip = (props: Props) => {
   const [hover, setHover] = useState(false);
+  const classes = useStyles();
 
   const handleToggled = (truncated: boolean) => {
     setHover(truncated);
   };
 
   return (
     <Tooltip
       title={props.title ?? props.text!}
       placement={props.placement}
       disableHoverListener={!hover}
     >
       <TextTruncate
         text={props.text}
         line={props.line}
         onToggled={handleToggled}
+        containerClassName={classes.container}
       />
     </Tooltip>
   );
 };

```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4602-5a1musfdott8leobiov0kxbu5waufoci.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/602febe565f8175876f4eb09

**Review by adamdmharvey — APPROVED** [review]
👏 looks great, thanks @Fox32 ! 👏

File: plugins/catalog/src/components/HasComponentsCard/HasComponentsCard.tsx [hunk 1]
```diff
@@ -33,6 +33,14 @@ type Props = {
   variant?: 'gridItem';
 };
 
+const columns = [
+  EntityTable.columns.createEntityRefColumn({ defaultKind: 'component' }),
```
Comment (freben) [inline plugins/catalog/src/components/HasComponentsCard/HasComponentsCard.tsx:37]: These are case insensitive right? It was all upper API above, just checking

**Review by freben — APPROVED** [review]
Sweet!

File: plugins/catalog/src/components/HasComponentsCard/HasComponentsCard.tsx (hunk 1 above)
Comment (Fox32) [reply plugins/catalog/src/components/HasComponentsCard/HasComponentsCard.tsx:37]: Yeah, I think it should be: https://github.com/backstage/backstage/blob/6c09312dcab36ea7d2e820aa5d1af65501b23c3e/plugins/catalog-react/src/components/EntityRefLink/format.ts#L50

**Pull request merged**